### PR TITLE
Workspaces pinned tabs

### PR DESCRIPTION
**Pull request by SvetozarMateev**
## New template - workspaces-react-pinned-tabs
Added a template that uses the new noTabHeader property and custom react buttons to showcase how can pinned tabs be implemented.

## New extension point in workspaces-react-ui
The components object now has a WorkspaceContents property which in combination with the `<WorkspaceContents/>` component provided by the library enables composition and conditional rendering of the workspace contents.

## Workspaces selected event
Added the `frame.onWorkspaceSelected()` event which is triggered when a workspace has been selected.

### CODE REVIEW

**Review by ggeorgievx — APPROVED** [review]

File: packages/workspaces-api/workspaces.d.ts [hunk 1]
```diff
@@ -182,6 +182,9 @@ export namespace Glue42Workspaces {
 
         /** States whether or not the workspace should have focus when opened. */
         isFocused?: boolean;
+
+        /** Provides the opportunity to open a workspace with a workspace tab header */
```
Comment (flashd2n) [inline packages/workspaces-api/workspaces.d.ts:186]: with **no** tab header

File: packages/workspaces-api/workspaces.d.ts [hunk 2]
```diff
@@ -403,6 +406,12 @@ export namespace Glue42Workspaces {
          */
         onWorkspaceOpened(callback: (workspace: Workspace) => void): Promise<Unsubscribe>;
 
+        /**
+         * Notifies when a workspace is selected in this frame and returns an unsubscribe function.
+         * @param callback Callback function to handle the event. Receives the added workspace as a parameter.
```
Comment (flashd2n) [inline packages/workspaces-api/workspaces.d.ts:411]: received the **selected** workspace

File: e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        }).then(unsub => {
+            unSubFuncs.push(unsub);
+            gtf.wait(3000, done);
+            return defaultFrame.createWorkspace(Object.assign(basicConfig, { config: { noTabHeader: true } }));
+        }).then((w) => {
+            newWorkspace = w;
+        }).catch(done);
+    });
+
+    it.only('not be invoked when a workspace with noTabHeader:true is restored', (done) => {
```
Comment (flashd2n) [inline e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js:108]: why .only

File: e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+                return firstWorkspace.focus();
+            }).catch(done);
+        });
+    });
+
+    it('not be invoked when a workspace with noTabHeader:true is opened', (done) => {
+        let newWorkspace = undefined;
+        defaultFrame.onWorkspaceSelected((w) => {
+            if (newWorkspace && w.id === newWorkspace.id) {
```
Comment (flashd2n) [inline e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js:96]: you don't need the if check here -> we should fail if the event is fired, no matter the args.

File: e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    it.only('not be invoked when a workspace with noTabHeader:true is restored', (done) => {
+        const layoutName = gtf.getWindowName("workspaceLayout");
+        let newWorkspace = undefined;
+
+        defaultFrame.workspaces().then((wsps) => {
+            const first = wsps[0];
+            return first.saveLayout(layoutName);
+        }).then(() => {
+            return defaultFrame.onWorkspaceSelected((w) => {
+                if (newWorkspace && w.id === newWorkspace.id) {
```
Comment (flashd2n) [inline e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js:117]: same as above -  no need for if statement

File: e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js [hunk 6]
```diff
@@ -0,0 +1,173 @@
+describe('frame.onWorkspaceSelected() Should ', () => {
```
Comment (flashd2n) [inline e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js:1]: every test needs to have the ready resolve condition (multiple ready invocations before marking the test as done)

File: e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+        }).then((w) => {
+            newWorkspace = w;
+        }).catch(done);
+    });
+
+    it('not be invoked when the unsub function is invoked and a workspace is focused', (done) => {
+        defaultFrame.workspaces().then((wsps) => {
+            const firstWorkspace = wsps[0];
+
+            defaultFrame.createWorkspace(basicConfig).then(() => {
```
Comment (flashd2n) [inline e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js:134]: unnecessary nesting. return the createWorkspace invocations and chain the then.

File: e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+                gtf.wait(3000, () => { done() });
+
+                return defaultFrame.onWorkspaceSelected((w) => {
+                    if (w.id === firstWorkspace.id) {
```
Comment (flashd2n) [inline e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js:138]: unnecessary if

**Review by flashd2n — CHANGES_REQUESTED** [review]

File: e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+        }).catch(done);
+    });
+
+    it('be invoked when a workspace is focused', (done) => {
+        let firstWorkspace = undefined;
+        const ready = gtf.waitFor(2, done);
+        defaultFrame.workspaces().then((wsps) => {
+            firstWorkspace = wsps[0];
+
+            defaultFrame.createWorkspace(basicConfig);
```
Comment (flashd2n) [inline e2e/tests/Workspaces/Frame/onWorkspaceSelected.spec.js:80]: missing return

**Review by flashd2n — CHANGES_REQUESTED** [review]

**Review by flashd2n — APPROVED** [review]

**Pull request merged**